warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java

```diff
@@ -2473,6 +2477,8 @@ public static GTSEncoder parse(GTSEncoder encoder, String str, Map<String,String
       throw new ParseException("Invalid timestamp.", tsoffset);
     }
 
+    timestamp += timeshift;
```

Comment:
There could be an overflow. Not sure if we should check and if we do, how does it impact performance.

Reply:
This is an edge case not worth checking at ingestion time. We do not do it for relative timestamps either (=T+..)